safenode/src/node/api.rs

```diff
@@ -186,12 +195,22 @@ impl Node {
     async fn handle_cmd(&mut self, cmd: Cmd) -> CmdResponse {
         match cmd {
             Cmd::StoreChunk(chunk) => {
-                let resp = self
-                    .chunks
-                    .store(&chunk)
-                    .await
-                    .map_err(ProtocolError::Storage);
-                CmdResponse::StoreChunk(resp)
+                debug!("That's a store chunk in for :{:?}", chunk.address().name());
+                // Create a Kademlia record for storage
+                let record = Record {
+                    key: RecordKey::new(chunk.address().name()),
+                    value: chunk.value().to_vec(),
+                    publisher: None,
+                    expires: None,
+                };
+
+                match self.network.regigster_as_provider_for_record(record).await {
+                    Ok(()) => CmdResponse::StoreChunk(Ok(())),
+                    Err(err) => {
+                        error!("Failed to register chunk as provider: {err:?}");
+                        CmdResponse::StoreChunk(Err(ProtocolError::ProvideRecordNotSaved))
```

Comment:
RecordNotSaved should be enough (or for stringency, Stored, since that's what we say in other places).

Otherwise, missing r at end of Provide.